### ISSUE TITLE
[mediaqueries-5] `prefers-contrast: high` media feature doesn't account for macOS and iOS

### ISSUE THREAD

**Issue by cookiecrook** [w3c/csswg-drafts#2943]
[mediaqueries-5] `prefers-contrast: high` media feature does not accurately describe the macOS/iOS state

https://drafts.csswg.org/mediaqueries-5/#prefers-contrast

Unlike Microsoft's "high contrast" mode, the "Increased Contrast" settings on macOS and iOS do not result in "high" contrast, but an intermediate "increased" value. It's possible that a future update to these and other systems would allow a user-end "high" contrast value in addition to the current implementation of "increased" contrast.

The media feature should account for values other than "high" and "low." Ideally there'd be a shorthand way to match both the macOS/iOS value, or the extreme end value used by Microsoft.

**Comment by cookiecrook** [comment]
A few possible variants to solve this:

1. Perhaps a numeric value, 0-100 where 50 is the default (or -100-100, where 0 is the default). There could be named values too. e.g. `high` = 100, `increase` = 60? The queries could include a grreater-than/less-than (</>) syntax such as:

````css
@media (prefers-contrast > 0) {}
@media (prefers-contrast > increase) { /* matches both 'increase' and 'high' values */ }
```

2. Have two separate media features that would allow use in a boolean context:
`prefers-increased-contrast: no-preference | increase | high` and `prefers-reduced-contrast: no-preference | reduce | low` 

```css
@media (prefers-increased-contrast)  { /* matches 'increase' or 'high' */ }
@media (prefers-reduced-contrast)  { /* matches 'reduce' or 'low' */ }
```

**Comment by Wolfr** [comment]
Just posting that I have an active need for this feature as a web dev trying to make websites more accessible. Not sure if this is the right place, I am new here.

My 2 cents on syntax, I would rather have a simpler `@media (prefers-increased-contrast) { }` without values since the surrounding implementation depends on the OS.

I know it is a slider in macOS but essentially it needs to map to a high contrast checkbox in the host OS (Windows/Mac/Linux). It is up to the OS to set good defaults to what it means. But to start working with numbers is too complex imo and will not drive usage in actual CSS.

**Comment by cookiecrook** [comment]
Another suggestion I recently discussed with @grorg:
	
```
prefers-contrast: [ reduce | no-preference | increase | high ]
```

In the boolean context `@media (prefers-contrast)` should match both `increase` or `high`, but not `reduce` or `no-preference`.

**Comment by cookiecrook** [comment]
@Wolfr wrote:
> I know it is a slider in macOS 

The legacy slider (rarely used) would not likely affect this media feature. Currently the thinking is to have it match the highly used "increase contrast" checkbox/switch on macOS or iOS...

**Comment by alice** [comment]
To recap, I think there are several distinct concepts of contrast preference:

- *True high contrast*, along the lines of the Windows High Contrast Mode. This is often a light-on-dark colour scheme with white or yellow text, black background, reduced detail, few gradations of colour etc. This is intended for users with unusual vision impairments, who may also use other accommodations such as screen magnification to be able to access content.
   - Despite typically being light-on-dark, this is quite different from a "night mode" theme.
   - This is often an operating system setting which affects how text, borders and images are displayed. In this case, developers may wish to make adjustments to ensure that these modifications don't create other problems. This is the "forced" case, I think.
  - However, certain apps also provide [high contrast themes](https://blog.jetbrains.com/idea/2018/10/intellij-idea-2018-3-eap-high-contrast-theme-and-more-accessibility-improvements/), and it would be good to be able to allow users to opt in to these themes on all sites which support a common theming mechanism, such as a media query.
- *Enhanced contrast*, along the lines of OS X's setting and Gmail's current high contrast theme, which retain much of the "standard" palette and level of detail but enhance borders and increase text contrast to around the WCAG AAA level or better. The WCAG AAA level is intended for users with moderate vision impairments with a visual acuity of 20/80-20/40.
- *Acceptable contrast*, which is denoted by [WCAG AA](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) level. This is intended to be a standard contrast level perceivable by people with 20/40 vision or better - considered "minimal vision impairment" and comprising around 10% of the population of a typical wealthy country.
- There is also a small subset users who prefer *reduced contrast* for various health reasons.

These seem to map nicely onto the levels in https://github.com/w3c/csswg-drafts/issues/2943#issuecomment-437613516 - although we seem to have lost the concept of "forced" from https://github.com/w3c/csswg-drafts/issues/443

I don't know that we'd want to conflate "high contrast" and "enhanced contrast", since they really look quite different and target different audiences. Perhaps they'd need to be separate boolean queries?

**Comment by cookiecrook** [comment]
@Alice wrote:

> although we seem to have lost the concept of "forced" from #443

That seems to be addressed sufficiently by the forced-colors media feature in the MQ5 draft.

**Comment by frivoal** [comment]
`true high contrast` mode, as MS does it, is forcing colors, often but not always to high contrast (it can be used to force low contrast, or any color scheme you want). This is primarily meant to be addressed through the `forced-colors` MQ together with [forced colors mode](https://drafts.csswg.org/css-color-adjust-1/#forced-colors-mode), though it does also trigger the `prefers-contrast: high` if the forced colors are indeed a high contrast set rather than low contrast one.

`prefers-contrast: high` alone, as I understand it, is more meant to reflect something along the lines of macOS's Enhanced Contrast.

Is there a situation where we have higher contrast than macOS's enhanced contrast, yet not have forced colors? If yes, it may be worth distinguishing between `prefers-contrast: increase` and `prefers-contrast: high`, but I have a bit of a hard time figuring what it is. If there isn't, what is the benefit of introducing the two levels on the `prefers-contrast` MQ in addition to having the `forced-colors` MQ?

**Comment by Wolfr** [comment]
From the author end it seems realistic to assign colors for 3 situations, but not more.

* Regular - the design as intended 
* Enhanced contrast (not the colors you wanted as an author but fixing them up to accord to a11y guidelines - and help users with this setting - while still not making them an extreme “other” version of the design)
* Max contrast (akin to Microsoft's almost black and white mode)

Most websites would implement (1) and (2) - the max contrast (3) would be more like a system thing.

**Comment by cookiecrook** [comment]
FYI that leaves out the "reduced" contrast value that some have requested, but I don't know of any native OS implementations of reduced contrast, so it might be okay to forego that value until a similar feature is adopted by at least one major OS.

**Comment by fuzzbomb** [comment]
@Wolfr 

> Max contrast (akin to Microsoft's almost black and white mode)  [...] would be more like a system thing.

That's what the `forced-colors` media query is for; it's not a author choice at all.

Note that Windows High-Contrast feature isn't always black and white. Users can specify any colours they want for a Windows High-Contrast theme... including _low contrast_ colours (e.g. brown text on a beige background, which suits some users with dyslexia). That's why it's being called `forced-colors` in the standards track, instead of "high/max contrast".

**Comment by cookiecrook** [comment]
> Is there a situation where we have higher contrast than macOS's enhanced contrast, yet not have forced colors? If yes, it may be worth distinguishing between prefers-contrast: increase and prefers-contrast: high, but I have a bit of a hard time figuring what it is. 

I am not aware of any current platform that supports multiple built-in levels of increased contrast, but it's possible for any platform to all them in the future. 

I'm trying to avoid having the value vocabulary locked to an extreme end ("high") which limits future flexibility: the syntax now implies platforms could only ever implement `no-preference` (default) or `high` contrast with nothing in-between. Furthermore, it does not accurately describe the current Mac/iOS setting, which is not a "high" contrast of around +10:1, but instead a "increased" contrast of +4.5:1. 

The same underlying technologies that allow Mac/iOS native and 3rd-party apps to support light and dark modes, also allow "increased" contrast color variants in each light and dark mode. There is nothing (other than a lot of engineering effort) technically stopping Mac/iOS from adding a non-forced "high contrast" mode (+10:1) using the same underlying system. If that were to happen, which CSS value should the existing "increased contrast" user match? It wouldn't make sense to match the `high` value, if the user and web author expectation of what "high" contrast means had changed.

A possible solution is to remove the current `high` value and replace it with a new one: `increase`. This could match either the Mac/iOS modes and the Windows mode today, but allow for future flexibility. If/when any platform shipped multiple contrast modes, the CSS WG could add the `high` value. At that point, Microsoft's current user setting could match both `increase` **or** `high`, but Mac/iOS's current user setting could match `increase` but **not** `high`.

**Comment by cookiecrook** [comment]
Updating the suggested solution summary:

**Option 1: Change the current value `high` to `increase`.** 

Forego additional values (`reduce` and `high`) until any native platform implements a feature that makes the distinction necessary. 

Con: Microsoft would need to remove their existing match for `high`.

```css
prefers-contrast: [ no-preference | increase ] /* remove `high` value */
```

**Option 2: Keep `high` value and add new value `increase`.** 

Forego the the low contrast value (`reduce`) until any native platform implements a feature that makes the distinction necessary. 

Con: Some authors may use the verbose version `prefers-contrast: high` and indavertently not match the Mac/iOS setting.
Pro: May allow authors to distinguish between the increase and high values.

```css
prefers-contrast: [ no-preference | increase | high ] 
```

**Common Usage: Same.**

In both proposals above, the 99.9% author usage case would be to use the boolean shorthand below, which would match all currently known increased or high contrast setting implementations.

```css
@media (prefers-contrast)
```

**Comment by ZekeMedley** [comment]
@cookiecrook 

> I don't know of any native OS implementations of reduced contrast, so it might be okay to forego that value until a similar feature is adopted by at least one major OS.

Unless someone has ideas about a system (or otherwise) setting that would indicate that the media query should be toggled to `low` this seems appropriate to me as well.

I'm looking into adding this to Firefox and despite some digging around am having trouble finding anything. If anyone has ideas though I'd love to hear them.

**Comment by frivoal** [comment]
A number of comments in #3856 also relate to this issue, so I encourage people to have a read (but please keep the discussion here).
In particular,  See also @cookiecrook 's comment in https://github.com/w3c/csswg-drafts/issues/3856#issuecomment-662028666, which includes:

> High Contrast (this is a 2m crummy comp, not an existing view) similar to Windows HC mode.
![comp of what a real high contrast mode might look like](https://user-images.githubusercontent.com/75867/88092448-5b1c8880-cb45-11ea-968c-9e43ec8454f9.png)

This is not really similar to the unfortunately named Windows High Contrast mode, since the Windows HC mode could result in that OR in a low contrast rendering, OR in an light-on-dark rendering, OR in a fuchsia-on-sepia rendering, or whatever the user wants. It is a forced color mode, and [we have an MQ for that](https://drafts.csswg.org/mediaqueries-5/#forced-colors) as well as more details in [https://drafts.csswg.org/css-color-adjust-1/#forced](css-color-adjust-1). Great presentation from @melanierichards here: https://www.w3.org/2019/09/Meetup/speaker-melanie.html

----

I continue not to understand why you refuse to implement the value in the spec which is defined to match the mode you have in macOS and insist that we introduce an additional value which maps to the mode you have in macOS.

Sure, there might be in the future an OS which has a high contrast mode that is neither that of macOS, nor is forced colors as seen on Windows, and needs to be distinguished from both of those. We'll deal with it when it exists and when we know how it actually works and to what degree it needs distinguishing. In the mean time, I don't see why we should dedicate a value in the spec to this hypothetical and add a value to deal with what actually exists today, instead of implementing the spec as is.

If we just want to rename `high` to `increased`, why not. It's more letters, so I'd rather not, but I don't care strongly. But I continue to claim that the current definition adequately describes what macOS does:

> Indicates that user has notified the system that they prefer an interface that has a higher level of contrast.

**Comment by melanierichards** [comment]
Forgive my naivete here, but is there a reason the group settled on keyword values instead of a numeric system?

I wonder if the numeric system should be explored again:

* Clear expectations for the author on what contrast level they should support (how does an author reason about `high` or `increase` values? What is "high" enough?), baked into the MQ itself
* Helps disambiguate from the Windows High Contrast feature
* I received feedback from a framework developer that they'd actually prefer to have numeric values for `prefers-contrast`, as they conceptualize contrast on a sliding scale; should open this up to the broader web dev community for input, of course.

I suppose the challenges may be 1) making sure numeric values don't explode in volume (maybe go off WCAG levels?) and 2) numeric values may not always align 1:1 with OS platform settings, but I think you'll get authors writing mis-aligned contrast ratios without clear guidance, anyway.

**Comment by alice** [comment]
> how does an author reason about high or increase values? What is "high" enough?

This is a great question, and I think gets to a lot of the confusion here. In particular, WCAG doesn't really give us any guidance for the "true high contrast" case, perhaps assuming that users who need it will be using a forced colors mode?

>  I don't see why we should dedicate a value in the spec to this hypothetical and add a value to deal with what actually exists today, instead of implementing the spec as is.

It's true that it is a hypothetical for the time being, but it's also true that the Windows High Contrast mode and Mac Increased Contrast mode are quite different and are meeting distinct user needs. 

A user who needs the level of contrast offered by Windows High Contrast default colours will not be able to use the Mac Increased Contrast mode, which is really more like a WCAG AAA mode in practice.

There are also variations in what websites which offer alternate colour themes are targeting.

Compare VS Code's high contrast theme (yes I know it's technically not a website):

![VS Code High Contrast - white text, black background, bright aqua borders, orange focus/selection highlights](https://user-images.githubusercontent.com/95208/88746119-1a4cf100-d18f-11ea-9995-2cb2543cc816.png)

with Gmail's "high" contrast theme, which is more like the Mac increased contrast mode:

![Gmail high contrast - darker grey text on lighter grey background, dark grey borders instead of light grey borders](https://user-images.githubusercontent.com/95208/88746198-4cf6e980-d18f-11ea-80bd-5bb6ea846bd0.png)

It may well be the case that "true" high contrast only happens in practice (other than VS Code) when there is a forced colors mode applying. However, I do also feel that we should avoid conflating these two modes, just as we avoid conflating high contrast and dark mode even though forced colours high contrast modes often do use a dark background.

A website which meets WCAG AAA probably wouldn't need to do anything (assuming they include sufficiently contrasting border colours in their design) in response to `prefers-contrast: increased`, but might need to adapt to a high contrast forced colors mode differently.

A separate question is whether enough websites would respect either `prefers-contrast: increased` or a non-forced `prefers-contrast: high` to make the feature valuable to users, compared to adding more forced colors modes.

**Comment by frivoal** [comment]
Summary of where I think we're at.

# Syntax 1 (spec today):

```
prefers-contrast: no-preference | low | high | forced
```

* MacOS's + iOS's “increase contrast” options map to `high`
* Microsoft's forced colors maps to `forced` (not `high`), and possibly, in addition, depending on the actual colors used, may map to `low` or `high` as well.

Authors should write `@media (prefers-contrast: high) { }` when writing high contrast styles.
`@media (prefers-contrast) { }` (without a specified value) only denotes that the author has some preference about contrast, but not which. An appropriate way to respond to this is to decrease visual complexity (replace background images with plain colors,  turn off decorative gradients, or replace border images or box shadows with simple solid borders…)

# Syntax 2

```
prefers-contrast: no-preference | low | increase | forced
```

Same as 1, just renaming `high` to `increase`, to avoid suggesting it is necessarily very high.
Downside: It's a tiny little bit longer.

# Syntax 3 [interpretation A]:

```
prefers-contrast: no-preference | low |  increase | high | forced
```


Distinguishes between high and increase.
MacOS/iOS would map to `increase`.
Microsoft Windows's forced colors still maps to `forced`, but now may map to `low`, `high`, **or `increase`** depending on the actual colors chosen?

Upside: This lets authors capture the difference between “somewhat high” and “very high”.
Downside: Writing `@media (prefers-contrast: increase) or (prefers-contrast:high) { }` is tedious.

# Syntax 3 [interpretation B]:

Same syntax as above, but inspired by how the `color-gamut` media feature works, any UA that matches `high` must also match `increase`, since high is higher than increase.
Authors can just write: `@media (prefers-contrast: increase)` when they don't want to distinguish.

# Syntax 4
```
prefers-low-contrast: no-preference | low
prefers-high-contrast: no-preference | increase | high
[forced-colors remains a separate query]
```


These are defined so that only one of `prefers-low-contrast` and `prefers-high-contrast` can be truthy at the same time.


Upside: `@media (prefers-high-contrast) {}` is less verbose than `@media (prefers-contrast: increase) or (prefers-contrast:high) { }`.
Downside: It's not actually less verbose than syntaxes 1, 2, or 3[B].
Downside: There's no succint way to detect all the preferred contrast modes.

# Key question

Syntaxes 3 and 4 allow distinguishing between `increase` and `high`, but syntaxes 3[B] and 4 provide shortcuts because usually you don't actually want to distinguish and it's annoying to make common case verbose. Do we actually really need to distinguish between “somewhat high” and “very high but not forced colors”? If not, I see no downside to syntax 1 or 2.

**Comment by alice** [comment]
Thanks for the comprehensive summary, @frivoal!

> `prefers-contrast: no-preference | low | high | forced`

Sorry for not having followed this discussion, but why is there both `prefers-contrast: forced` and `forced-colors`? When would the former apply but not the latter?

> Microsoft Windows's forced colors still maps to `forced`, but now may map to `low`, `high`, or `increase` depending on the actual colors chosen?

I definitely agree that we shouldn't match `prefers-contrast: high` if a user has selected low or moderate contrast colours.

I am curious to understand the usage modes of Windows High Contrast Mode (WHCM) in practice. I recognise that the colours are user-configurable, but to what extent are users modifying the colours chosen to enable a contrast preference other than "high" (along the lines of the extremely contrasting default values of black background and fully saturated yellow/cyan/green/white foreground colours)? I would like to better understand the user needs behind choices of low-contrast and increased-contrast colour options for WHCM.

> Do we actually really need to distinguish between “somewhat high” and “very high but not forced colors”?

This captures the key question extremely well.

My feeling is that the answer is yes, we do need to distinguish between those two concepts, because the user needs and the resulting accommodations are quite distinct from one another. In particular, a site or theme which meets WCAG AAA would likely be suitable for "increased" but likely not for "high" contrast.

Even though it's not currently possible for a user on Mac to enable a "high" contrast preference (or "low" for that matter), or for a user on Windows to enable an "increased" contrast preference, I think leaving scope for those preferences to be added would be preferable to conflating these two modes.

Whether we are likely to ever have a "very high but not forced colours" preference is debatable, but even if we don't, I think it's still helpful to understand that the colours that have been chosen represent a high contrast preference (or low contrast, as the case may be) for things like icon images.

**Comment by frivoal** [comment]
> why is there both prefers-contrast: forced and forced-colors? When would the former apply but not the latter?

They are expected/specified to both match at the same time.

The idea to have `prefers-contrast: forced` was because in addition to actually adjusting contrast or changing colors, a common (and generally desirable) response to either contrast preferences and to forced colors is to reduce visual clutter and to tone down decorative things (e.g. replacing background images with plain colors, turning off decorative gradients, or replacing border images or box shadows with simple solid borders…), and so being able to target all these with a common and simple query (`@media (prefers-contrast)`) was seen as desirable.

The reason for having that in addition to, rather than instead of, `forced-colors` is that `forced-colors` has already shipped (in Edge).

Overall, whether one likes that aspect or or not, I think it is mostly orthogonal to the rest of the discussion here.

> to what extent are users modifying the colours chosen to enable a contrast preference other than "high"

I don't have statistics, but I know of at least one case: some people with dyslexia find contrasty things hard to read, as they feel that the letters shine / sparkle as if backlit by too bright a light, and find low contrast to be more comfortable.

As for not necessarily accessibility centric uses for `low`, I'd say the popularity usage of sepia tones in ebook-reader (or reader mode) is an interesting indicator.

> >  Do we actually really need to distinguish between “somewhat high” and “very high but not forced colors”?
>
> This captures the key question extremely well.
> 
> My feeling is that the answer is yes […]

I agree that a very high contrast design, like that of VS Code, isn't the same as an enhanced contrast design, like the one of gmail. But also, I think it the number of sites that intent to have an enhanced contrast design in the manner of macOS/gmail, and to respond to forced colors, and to have on top of that a very high contrast mode separate from the forced color mode, is likely very small. Note for instance that VS Code and gmail only have one of each, not both. It is not a theoretical impossibility, but it is not a common case, and no OS I know of today offers users a setting to express that.

This is also part of the tension that I see in some earlier proposals: they suggest we need to distinguish between the two and therefore have two keywords, but also, since most of the time authors would want to query about both of them together and not distinguish, we need to have a syntax that's not too verbose for doing that.

So, if we want a syntax that's not too verbose for the common cases, as well the ability for authors (but not the obligation) to distinguish between high and very high contrast, I'd say something along the lines of 3B from https://github.com/w3c/csswg-drafts/issues/2943#issuecomment-665453076 is likely to be a good compromise. Right now, no OS ships the very-high-but-not-forced setting, so I'd be tempted to put it off until someone does ship it, but to solidify the design, we could also include it as an at-risk feature. And for the sake of brevity, I'd probably go with `high | boost` or `high | max` rather than `increase | high`

So, in summary, that means we'd have something like this:
> ```
> prefers-contrast: no-preference | forced | low | high | max
> ```
> <dl>
>  […]
> <dt>high
> <dd>Indicates that user has notified the system that they prefer an interface that has a higher level of contrast.
> <dt>max
> <dd>Indicates that the user has notified the system that they prefer an extremely high contrast, such as a black background with fully saturated yellow/cyan/green/white foreground colors (or vice versa, see `prefers-color-scheme` for information on which is prefered). User Agents that match this value must also simultaneously match `high`.
> </dl>

**Comment by alice** [comment]
Thanks for the explanation on `forced-colors`.

>  I think it the number of sites that intent to have an enhanced contrast design in the manner of macOS/gmail, and to respond to forced colors, and to have on top of that a very high contrast mode separate from the forced color mode, is likely very small.

I think where sites already have multiple themes, it actually wouldn't be a stretch to imagine them having both an "increased" contrast and a "high" contrast theme[1]. 

That said, I agree that forcing authors to type `@media (prefers-contrast: increase) or (prefers-contrast:high)` is probably best avoided.

Honestly, though, I don't know that they should be matching both in any case. I am not sure either way whether users who prefer AAA-level contrast would prefer to see VS Code style high contrast compared to a baseline experience. I think that is a question we should maybe look into[1].

If it turns out they do, then something along the lines of your last proposal here would work well; otherwise, we can just leave out the last sentence. I would prefer to avoid conflating the ideas of enhanced/AAA contrast and high contrast if we can, but as long as we can carefully document that "high" is shorthand for "increased", I can probably live with it.

[1] with the obvious caveat that sites have no incentive to provide a true high contrast theme, since there is no automatic way for a user to opt in to it without also having forced colours.

**Comment by frivoal** [comment]
> Honestly, though, I don't know that they should be matching both in any case. I am not sure either way whether users who prefer AAA-level contrast would prefer to see VS Code style high contrast compared to a baseline experience. I think that is a question we should maybe look into[1].

No, but the other way around is useful, and that's what the syntax I suggested would achieve. When users set their preference for VSCode-like "max contrast", if a site doesn't have that theme, but does have the gmail-like "high constrast", getting at least that is better than nothing. so gmail queries for "high", and that matches both when the actual preference is "high" and when its "max". But VSCode specifically queries for "max", and so it only turns on that mode for people who prefer max.

**Comment by astearns** [comment]
@cookiecrook @alice would you be able to join us in next week's meeting at https://www.timeanddate.com/worldclock/fixedtime.html?iso=20200805T23 ?

**Comment by cookiecrook** [comment]
I’m out on vacation (and without reliable WiFi) until the week of the August 10th.

**Comment by astearns** [comment]
Hmm. Then the next meeting where you could attend would be the 12th, which is at a bad time for Alice ( https://www.timeanddate.com/worldclock/fixedtime.html?iso=20200812T1 ). The next APAC-timed call would be Sept 2.

**Comment by tabatkins** [comment]
(Jumping back to just answer this one question.)

> Forgive my naivete here (and broadening the scope of the discussion), but is there a reason the group settled on keyword values instead of a numeric system?

In general, MQs aren't well-suited for fine-grained distinctions between values. Responding to an MQ is a fairly "heavyweight" operation generally, and having a lot of potential breakpoints leads authors to either feel like they should be responding to a bunch of points along a gradient (making the author do a lot of likely-unnecessary work), or responding to just a few chosen in an arbitrary way that's not likely to be consistent across sites (bad for the user).

The rare exceptions, like the screen size queries, are reasonable because they correspond directly to things the designer already cares about, *and* are trivially testable by just dragging the window larger/smaller.

But for most things, we serve both the author and the user better by creating a small, opinionated set of values to respond to - it guides the author in how much effort they should put in (one unit of work per value, basically), and gives the user a more consistent treatment across sites, since everyone will be responding to the same single value near the user's actual preference.

**Comment by alice** [comment]
Apologies for being so slow to respond - I was on vacation without wifi all last week.

I think you can go ahead without me - I can sync up with James beforehand to make sure we're on the same page.

**Comment by cookiecrook** [comment]
@frivoal wrote:
> When users set their preference for VSCode-like "max contrast", if a site doesn't have that theme, but does have the gmail-like "high contrast", getting at least that is better than nothing. so gmail queries for "high", and that matches both when the actual preference is "high" and when its "max". But VSCode specifically queries for "max", and so it only turns on that mode for people who prefer max.

I seems like the main disagreement is with the vocabulary. We're using the term "high" differently. 

To me, "high" contrast means ~"max or nearly max contrast"… iOS and macOS do not have a system setting for that (it's "increased" a.k.a. "higher, but not high"), but both those platforms _could_ have a true "high" or "max" setting, so I'm hoping we can avoid using the term "high." 

I wouldn't have thought "high" contrast would be a subjective term, but as it's clear several of us are interpreting it differently, it's likely that authors will too.

**Comment by cookiecrook** [comment]
The other main issue I've taken with the current syntax proposal is that IMO, too many tangentially related concepts are being squeezed into a single property: `prefers-contrast`.

The `forced` value has no direct relationship to contrast. Forced colors can be any value, resulting in higher, lower, or no change in contrast. Granted, Windows has a few "max" contrast themes that are also forced, but in my mind, the `forced` concept is closer to `prefers-color-scheme` than `prefers-contrast.` Why shoehorn it in here when it already has its own property: `forced-colors`?

**Comment by cookiecrook** [comment]
Likewise, the addition of `low` makes authoring less convenient since the boolean match is now ambiguous. I'm also unaware of any planned or existing system setting that would result in a match for `low`. The sepia theme Florian mentioned could be addressed in `forced-colors` or 'prefers-color-scheme.`

**Comment by cookiecrook** [comment]
I'm available at [2020-08-12 T 16:00 GMT](https://www.timeanddate.com/worldclock/fixedtime.html?iso=20200812T16) (this Wednesday at 9 AM Pacific) to discuss. 

Ideally I'd like to see a useful boolean match (or otherwise very short authoring match) for `prefers-contrast`… I think this could be achieved a few different ways:

Updating my proposal list from above: 

**Option 1: Change the current value `high` to `increase`.**

```css
prefers-contrast: [ no-preference | increase ] /* the `increase` value can match both the iOS/macOS "increased" contrast and Windows "high" or "max" contrast settings. */
```

**Option 2: Change `high` value to `max` and add new value `increase`.**

Allows distinction between the different settings, but authors would likely 

```css
prefers-contrast: [ no-preference | increase | max ] 
```

Forego `low` or `decrease` value until any native platform implements a feature that makes the distinction from `forced-colors` necessary.

**Common Usage: Same.**

In both proposals above, the 99.9% author usage case would be to use the boolean shorthand below, which would match all currently known "increased" or "max" contrast setting implementations.

```css
@media (prefers-contrast)
```

**Comment by cookiecrook** [comment]
Another possibility that would allow the `low` or `decreased` value is to associate the named values with a linear numeric scale like Meredith mentioned, then using the greater-than/less-than syntax:

```css
@media (prefers-contrast > increase) { } /* matches `increase` or `max` but not `low` or `no-preference` */
@media (prefers-contrast < max) { } /* matches `low`/`decrease`, `no-preference`, or `increase` but not `max` */
```

I wanted to mention the possibilities above, though these are less desirable to me because of:
- the added complexity in general, and
- the boolean would match the `low` value... therefore `@media (prefers-contrast)` could mean the opposite: "does NOT prefer contrast."

**Comment by tabatkins** [comment]
> the boolean would match the low value...

Yeah, English is a little hard... It's accurate to say that the boolean matching means "has a contrast preference", just not which way the preference leans. In general, the "true" boolean match for the prefers-* MQs isn't a particularly useful value, only the "false" match is. (It's just that several of the prefers-* only have a single truthy value, so an author can pretend that the presence of a preference tells them exactly what the preference is.)

**Comment by cookiecrook** [comment]
> In general, the "true" boolean match for the prefers-* MQs isn't a particularly useful value, only the "false" match is. 

I don't agree that this could never be useful. There's a lot of opportunity to have the boolean match a minimum simple preference while leaving an opening for more specific values. I haven't pushed for the more specific values yet because MQ would need to address anti-fingerprinting security before the more specific values should be implemented.

For example, iOS is already shipping two variants of a native "reduce motion" setting that differentiates between pans (e.g. automatic side scrolls between screens) and other types of problematic vestibular motion like parallax and scale/zoom. The interface pans often convey a hierarchy. Other, more decorative animations are usually more extraneous, and often more problematic for those with vestibular disorders.

One could see a future expansion of `prefers-*` to one or more optional variants, making the boolean match most useful for authors.

```css
prefers-reduced-motion: no-preference | [ reduce | [exclude-parallax &| exclude-scale &| … ]
prefers-reduced-opacity: no-preference | [ reduce | opaque ]
prefers-(increased)-contrast: no-preference | [ increase |  max ]
prefers-color-scheme: light | dark | forced
```

**Comment by cookiecrook** [comment]
I think this is a reasonable, simple solution because:

1. It keeps the common use case short:

```css
@media (prefers-contrast: more)
```

2. It keeps the syntax open to other values like 'max` (non-forced, extremely high) if they are needed in future iterations.

The rest of the points I raised in this thread—such as the duplicated: "prefers-contrast: forced" and "forced-colors: active"— can be raised as separate issues.

**Comment by cookiecrook** [comment]
The working group also agreed to add more use cases via request from @Alice.
